src/MOI_wrapper.jl

```diff
@@ -260,20 +260,19 @@ end
 # coef: List of corresponding coefficients
 # d: dimension of set
 # rev: if true, we unscale instead (e.g. divide by √2 instead of multiply for PSD cone)
-function _scalecoef(rows, coef, d, rev)
+function _scalecoef(rows::Union{UnitRange{Int64}, Vector{Int64}}, coef::Vector{Float64}, d::Int64, rev::Bool)
     scaling = rev ? 1 / √2 : 1 * √2
     output = copy(coef)
-    diagidx = BitSet()
-    for i in 1:d
-        push!(diagidx, trimap(i, i))
-    end
     for i in 1:length(output)
-        if !(rows[i] in diagidx)
+        # See https://en.wikipedia.org/wiki/Triangular_number#Triangular_roots_and_tests_for_triangular_numbers
+        is_diagonal_index = isinteger(sqrt(8*rows[i] + 1))
```

Comment:
Is `sqrt` guaranteed to return an integer for an input that is a perfect square ? Why not use `isqrt` instead ? It seems to be also faster:
```julia
julia> f(x) = @benchmark isqrt($x)^2 == $x
f (generic function with 1 method)

julia> f(49)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     9.593 ns (0.00% GC)
  median time:      10.280 ns (0.00% GC)
  mean time:        11.226 ns (0.00% GC)
  maximum time:     54.609 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     999

julia> f(49)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     9.554 ns (0.00% GC)
  median time:      9.695 ns (0.00% GC)
  mean time:        10.486 ns (0.00% GC)
  maximum time:     40.288 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     999

julia> f(49)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     9.555 ns (0.00% GC)
  median time:      9.694 ns (0.00% GC)
  mean time:        10.496 ns (0.00% GC)
  maximum time:     41.425 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     999

julia> f(x) = @benchmark isinteger(sqrt($x))
f (generic function with 1 method)

julia> f(49)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     10.894 ns (0.00% GC)
  median time:      11.028 ns (0.00% GC)
  mean time:        11.925 ns (0.00% GC)
  maximum time:     43.504 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     998

julia> f(49)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     10.894 ns (0.00% GC)
  median time:      11.019 ns (0.00% GC)
  mean time:        11.779 ns (0.00% GC)
  maximum time:     55.818 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     998

julia> f(49)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     10.895 ns (0.00% GC)
  median time:      11.020 ns (0.00% GC)
  mean time:        11.748 ns (0.00% GC)
  maximum time:     31.319 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     998
```

Reply:
Yes, will change this accordingly.